erica_app/erica/elster_xml/elster_xml_generator.py

```diff
@@ -109,7 +109,7 @@ def generate_full_xml(th_fields, nutzdaten_header_generator, nutzdaten_generator
 
 
 def generate_full_est_xml(form_data, steuernummer, year, person_a_idnr, first_name, last_name, street, street_nr, plz,
-                          town, empfaenger, nutzdaten_ticket="1", th_fields=None, use_testmerker=False,
+                          town, empfaenger, nutzdaten_ticket="1", new_admission=False, th_fields=None, use_testmerker=False,
```

Comment:
Adding all these boolean flags one by one is a bit of a code smell to me: they introduce a lot of tangled/coupled code, which can be hard to maintain (see https://martinfowler.com/bliki/FlagArgument.html for a more eloquent problem description).

I don't see a good way yet to split this into separate functions, do you? Maybe a small step could be to extract a `_add_tax_nr(vorsatz, new_admission)` function from `_generate_vorsatz` to be called on line 127?

Reply:
Yes, I've added the flags only because I wanted to make it explicit. 

So, basically something like:
```python
     vorsatz = _generate_vorsatz(steuernummer, year, person_a_idnr, person_b_idnr, first_name, last_name, street,
                                 street_nr, plz, town)
                                 street_nr, plz, town)
     if new_admission:
          _add_tax_nr(vorsatz, new_admission) 
     _add_xml_nutzdaten_header(nutzdaten_block_xml, nutzdaten_ticket, empfaenger)
```

The other way would be to take `generate_vorsatz` out of this function and have vorsatz as parameter. We do not need the xml for the vorsatz. So, this could happen anywhere else. And then we could have two functions for vorsatz: `generate_vorsatz_with_tax_nr` and `generate_vorsatz_without_tax_nr`

Reply:
I like your second suggestion!